src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java

```diff
@@ -352,7 +352,7 @@ private void addAdditionalDataFromManagerToJson(AdditionalDataManager additional
 	 * @return			the derived class if found, or null if not applicable
 	 */
 	@Nullable
-	public Class<?> getDerivedClass(@Nonnull final JsonObject jsonObject, @Nonnull final Class<?> parentClass) {
+	public Class<?> getDerivedClass(@Nonnull final JsonObject jsonObject, @Nullable final Class<?> parentClass) {
```

Comment:
Probably off-topic but probably also doesn't hurt: if a parameter should be null ever (i.e. marked with Nonnull), then it might be a good practice to check that explicitly in the method (e.g. Objects.requireNonNull())

Reply:
yep, I'm planning to do a bulk pass for that as part of this static analysis effort. We didn't use to have any annotations not long ago...

Reply:
see the latest commits, I added a lot of defensive programming for nonnull annotations. Also corrected a few annotations that should've been Nullable instead